components/raindrop/raindrop.app.mjs

```diff
@@ -1,11 +1,167 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "raindrop",
-  propDefinitions: {},
+  propDefinitions: {
+    raindropID: {
+      type: "string",
+      label: "Raindrop ID",
+      description: "Existing raindrop ID",
+    },
+    collectionId: {
+      type: "string",
+      label: "Collection ID",
+      description: "The collection ID",
+      async options() {
+        const { items } = await this.getCollections();
+        return items.map((e) => ({
+          value: e._id,
+          label: e.title,
+        }));
+      },
+    },
+    expanded: {
+      type: "boolean",
+      label: "Expanded",
+      description: "Whether the collection's sub-collections are expanded",
+      optional: true,
+    },
+    title: {
+      type: "string",
+      label: "Title",
+      description: "Name of the collection",
+    },
+    sort: {
+      type: "integer",
+      label: "Sort",
+      description: "The order of collection (descending). Defines the position of the collection among all the collections with the same parent ID",
+      optional: true,
+    },
+    public: {
+      type: "boolean",
+      label: "Public",
+      description: "Collection and raindrops that it contains will be accessible without authentication?",
+      optional: true,
+    },
+    view: {
+      type: "string",
+      label: "View",
+      description: "View style of collection. Defaults to list",
+      optional: true,
+      options() {
+        return [
+          "list",
+          "simple",
+          "grid",
+          "masonry",
+        ];
+      },
+    },
+    cover: {
+      type: "string[]",
+      label: "Cover",
+      description: "Collection cover url",
+      optional: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async _makeRequest($ = this, opts) {
+      const {
+        method = "get",
+        path,
+        data,
+        params,
+        ...otherOpts
+      } = opts;
+      return axios($, {
+        method,
+        url: `https://api.raindrop.io/rest/v1${path}`,
+        headers: {
+          ...opts.headers,
+          "user-agent": "@PipedreamHQ/pipedream v0.1",
+          ...this._getHeaders(),
+        },
+        data,
+        params,
+        ...otherOpts,
+      });
+    },
+    _getHeaders() {
+      return {
+        Authorization: `Bearer ${this.$auth.oauth_access_token}`,
+      };
+    },
```

Comment:
Since `_makeRequest` is the only place that `_getHeaders` is being used, could we eliminate it?
```suggestion
        headers: {
          ...opts.headers,
          "user-agent": "@PipedreamHQ/pipedream v0.1",
          Authorization: `Bearer ${this.$auth.oauth_access_token}`,
        },
        data,
        params,
        ...otherOpts,
      });
    },
```